### PR TITLE
simplify SslStream_StreamToStream_Alpn_NonMatchingProtocols_Fail test

### PR DESCRIPTION
**Pull request by wfurt**
- use TestHelper to get streams
- There is no need for SupportsAlpnAlerts: SslStream will inject Alert if underlying implementation does not

### CODE REVIEW

File: src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamAlpnTests.cs [hunk 1]
```diff
@@ -132,65 +132,49 @@ public async Task SslStream_StreamToStream_Alpn_Success(List<SslApplicationProto
...
-                        SslClientAuthenticationOptions clientOptions = new SslClientAuthenticationOptions
-                        {
-                            ApplicationProtocols = new List<SslApplicationProtocol> { SslApplicationProtocol.Http11 },
-                            RemoteCertificateValidationCallback = AllowAnyServerCertificate,
-                            TargetHost = certificate.GetNameInfo(X509NameType.SimpleName, false),
-                        };
+                        await serverStream.AuthenticateAsServerAsync(TestAuthenticateAsync, serverOptions);
+                        Assert.True(false, "AuthenticationException was not thrown.");
+                    }
+                    catch (AuthenticationException) { serverStream.Dispose(); }
```
Comment (stephentoub) [inline src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamAlpnTests.cs:163]:
```C#
await Assert.ThrowsAsync<AuthenticationException>(() => serverStream.AuthenticateAsServerAsync(TestAuthenticateAsync, serverOptions));
serverStream.Dispose();
```
?

File: src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamAlpnTests.cs [hunk 2]
```diff
@@ -132,65 +132,49 @@ public async Task SslStream_StreamToStream_Alpn_Success(List<SslApplicationProto
...
-                            }
-                            catch (AuthenticationException) { server.Dispose(); }
-
-                            await TestConfiguration.WhenAllOrAnyFailedWithTimeout(t1);
-                        }
-                        else
-                        {
-                            Task t1 = clientStream.AuthenticateAsClientAsync(TestAuthenticateAsync, clientOptions);
-                            Task t2 = serverStream.AuthenticateAsServerAsync(TestAuthenticateAsync, serverOptions);
+                    await TestConfiguration.WhenAllOrAnyFailedWithTimeout(t1);
```
Comment (stephentoub) [inline src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamAlpnTests.cs:165]: WhenAllOrAnyFailedWithTimeout makes little sense when passed a single task :)

Comment (wfurt) [reply src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamAlpnTests.cs:165]: yes. that was blind copy from the old test.

Comment (wfurt) [reply src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamAlpnTests.cs:165]: I could not find WithTimeout, what do I miss? I saw proposal but that did not look like ready.

Comment (stephentoub) [reply src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamAlpnTests.cs:165]: In this library the test helper is actually called TimeoutAfter.

Comment (wfurt) [reply src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamAlpnTests.cs]: updated. Would it make sense to make it consistent with other assemblies? (or wait for  #27723) ?

Comment (stephentoub) [reply src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamAlpnTests.cs]: > Would it make sense to make it consistent with other assemblies?

Yup

**Review by stephentoub — APPROVED** [review]

**Pull request merged**